src/execution/index/fixed_size_allocator.cpp

```diff
@@ -253,6 +253,12 @@ FixedSizeAllocatorInfo FixedSizeAllocator::GetInfo() const {
 
 	for (const auto &buffer : buffers) {
 		info.buffer_ids.push_back(buffer.first);
+
+		// Memory safety check.
+		if (buffer.first > idx_t(MAX_ROW_ID)) {
```

Comment:
Is there any more strict check we can use here? e.g. the actual amount of buffers that we have?